synapse/module_api/errors.py

```diff
@@ -16,14 +16,18 @@
 
 from synapse.api.errors import (
     InvalidClientCredentialsError,
+    NotFoundError,
     RedirectException,
     SynapseError,
 )
 from synapse.config._base import ConfigError
+from synapse.handlers.push_rules import InvalidRuleException
 
 __all__ = [
     "InvalidClientCredentialsError",
     "RedirectException",
     "SynapseError",
     "ConfigError",
+    "InvalidRuleException",
+    "NotFoundError",
```

Comment:
ugh, I'm not that enthusiastic about exposing NotFoundError. It's a bit of a poster-child for our antipattern of raising HTTP errors from the depths of database code.

I see we already have a `RuleNotFoundException`. Can we use that?